### PR TITLE
Hrcpp 377/sasl conf

### PR DESCRIPTION
**Pull request by rigazilla**
SASL (PLAIN, MD5) for .NET.

Ref: https://issues.jboss.org/browse/HRCPP-377

### CODE REVIEW

File: src/main/cs/Infinispan/HotRod/Config/AuthenticationConfigurationBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            jniBuilder.ServerFQDN(serverFQDN);
+            return this;
+        }
+
+        void setupCallback(Infinispan.HotRod.SWIGGen.SaslCallbackHandlerMap map)
+        {
+            jniBuilder.setupCallback(map);
+        }
+
+        public AuthenticationConfigurationBuilder setupCallback(System.Collections.Generic.IDictionary<int, AuthenticationStringCallback> map)
```
Comment (mgencur) [inline src/main/cs/Infinispan/HotRod/Config/AuthenticationConfigurationBuilder.cs]: setupCallback -> SetupCallback

File: src/main/cs/Infinispan/HotRod/Config/ConfigurationBuilder.cs [hunk 2]
```diff
@@ -85,6 +85,11 @@ public SslConfigurationBuilder Ssl()
             return new SslConfigurationBuilder(this, builder.Ssl());
         }
 
+        public SecurityConfigurationBuilder Security()
```
Comment (mgencur) [inline src/main/cs/Infinispan/HotRod/Config/ConfigurationBuilder.cs]: So now we have Security at the same level as Ssl(). The java client has it differently: Security at the top level, and Authentication and Ssl under it. Ssl should probably be under Security.

File: src/main/cs/Infinispan/HotRod/Config/AuthenticationConfigurationBuilder.cs (hunk 1 above)
Comment (rigazilla) [inline src/main/cs/Infinispan/HotRod/Config/AuthenticationConfigurationBuilder.cs]: fixed

File: src/main/cs/Infinispan/HotRod/Config/ConfigurationBuilder.cs (hunk 2 above)
Comment (rigazilla) [inline src/main/cs/Infinispan/HotRod/Config/ConfigurationBuilder.cs]: You are right, but I can't change the interface in a minor release.

**Comment by mgencur** [comment]
Integrated.

**Pull request merged**